File: app-core/src/main/java/ru/micron/persistence/model/FilmReceive.java
package ru.micron.persistence.model;

import java.util.Objects;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.Hibernate;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "film_receive")
public class FilmReceive {

  @Id
  @Column(name = "imdb_id")
  private String imdbId;

  @Column(name = "is_received")
  private Boolean isReceived;

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) {
      return false;
    }
    FilmReceive that = (FilmReceive) o;
    return imdbId != null && Objects.equals(imdbId, that.imdbId);
  }

  @Override
  public int hashCode() {
    return getClass().hashCode();
  }
}

File: app-core/src/main/kotlin/ru/micron/persistence/model/FilmReceive.kt
package ru.micron.persistence.model

import org.hibernate.Hibernate
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "film_receive")
data class FilmReceive(
    @Id
    @Column(name = "imdb_id", nullable = false)
    var imdbId: String? = null,

    @Column(name = "is_received", nullable = false)
    var isReceived: Boolean? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) {
            return false
        }
        val that = other as FilmReceive
        return imdbId != null && imdbId == that.imdbId
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}